### PR TITLE
Assert redirects

### LINKED ISSUE #69: Assert login redirect fixture
_Originally posted by @AdrianDAlessandro in https://github.com/ImperialCollegeLondon/dune_processes/pull/64#discussion_r1756659179_

This should check it is redirected to the right place. Maybe using [`assertRedirects`](https://docs.djangoproject.com/en/5.1/topics/testing/tools/#django.test.SimpleTestCase.assertRedirects)?

Also, we will want to put this bit of code in every view that requires an authenticated user. My first instinct is to use a base test class that does this check. Something like:

```python
class LoginRequiredTest:
    endpoint: str

    def test_login_redirect(self, client):
        response = client.get(self.endpoint)
        assertRedirects(response, reverse("main:login"))

class TestIndex(LoginRequiredTest):
    endpoint = reverse("main:index")
    # the rest of the tests for the index view
```

An alternative is using pytest fixtures in an intelligent way. One option is to [pass data to fixtures](https://docs.pytest.org/en/latest/how-to/fixtures.html#using-markers-to-pass-data-to-fixtures) and run the check in that fixture.

Perhaps the best way is to leverage this [Factories as fixtures](https://docs.pytest.org/en/latest/how-to/fixtures.html#factories-as-fixtures) behaviour to create an `assert_login_redirect` fixture that can be called inside the test. But I don't think that's as explicit as inheriting from a base test class

### PR DESCRIPTION
**Pull request by dalonsoa**
# Description

Adds base test classes - and then actual test classes - to assert the page redirection when visiting views that require authentication. 

Fixes #69 

## Type of change

- [ ] Documentation (non-breaking change that adds or improves the documentation)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Optimization (non-breaking, back-end change that speeds up the code)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] Breaking change (whatever its nature)

## Key checklist

- [x] All tests pass (eg. `python -m pytest`)
- [x] The documentation builds and looks OK (eg. `python -m sphinx -b html docs docs/build`)
- [x] Pre-commit hooks run successfully (eg. `pre-commit run --all-files`)

## Further checks

- [x] Code is commented, particularly in hard-to-understand areas
- [x] Tests added or an issue has been opened to tackle that in the future. (Indicate issue here: # (issue))

### CODE REVIEW

**Review by cc-a — APPROVED** [review]
Looks nice. Thanks for expanding out test coverage. I've suggested a few minor simplifications.

File: tests/test_views.py [hunk 1]
```diff
@@ -3,76 +3,112 @@
...
+
+
+class TestLogsView(LoginRequiredTest):
+    """Tests for the logs view."""
+
+    @classmethod
+    def setup_class(cls):
+        """Set up the endpoint for the tests."""
+        cls.uuid = uuid4()
+        cls.endpoint = reverse("main:logs", kwargs=dict(uuid=cls.uuid))
```
[the remark this answers is not in the document]
Comment (dalonsoa) [reply tests/test_views.py:76]: Nope. We need the `uuid` in one of the tests. See line 85

[the remark this answers is not in the document]
Comment (dalonsoa) [reply tests/test_views.py:74]: Sorry, I mean I tried this, but it was unhappy about it. Let me check again, in case I did something wrong.

**Comment by dalonsoa** [comment]
All good!

**Pull request merged**